bugsnag-plugin-android-ndk/src/main/jni/bugsnag_ndk.c

```diff
@@ -151,7 +166,20 @@ Java_com_bugsnag_android_ndk_NativeBridge_deliverReportAtPath(
   }
   remove(report_path);
   (*env)->ReleaseStringUTFChars(env, _report_path, report_path);
+  goto exit;
+
+  exit:
   pthread_mutex_unlock(&bsg_native_delivery_mutex);
+  if (jpayload != NULL) {
+    (*env)->ReleaseByteArrayElements(env, jpayload, (jbyte *)payload,
+                                     0); // <-- frees payload
+  }
+  if (jstage != NULL) {
+    (*env)->ReleaseByteArrayElements(
+        env, jstage, (jbyte *)report->app.release_stage, JNI_COMMIT);
+  }
+  (*env)->DeleteLocalRef(env, jpayload);
+  (*env)->DeleteLocalRef(env, jstage);
```

Comment:
Updated implementation so that `report` is freed. Also created a PR for v5 as this also affects the latest version: https://github.com/bugsnag/bugsnag-android/pull/1140